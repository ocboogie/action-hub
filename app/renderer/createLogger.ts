// tslint:disable-next-line no-implicit-dependencies
import { ipcRenderer } from "electron";
import * as toastify from "react-toastify";

import Logger, { ILoggerObj, IMessage } from "../common/Logger";

export default (loggerObj: ILoggerObj): Logger => {
  const logger = Logger.fromObj(loggerObj);

  logger.addGlobalListener((category: string, msg: IMessage) => {
    Logger.log2Console(category, msg);
    toastify.toast(
      msg.title +
        (msg.description
          ? ". Look at the console for more info ctrl+shift+i"
          : ""),
      {
        type: category as toastify.ToastType
      }
    );
    ipcRenderer.send("clean-history");
  });

  // Catch up
  logger.history.forEach(logged => {
    logger.report(logged.category, logged.msg.title, logged.msg.description);
  });

  ipcRenderer.on(
    "log",
    (event: Electron.Event, category: string, msg: IMessage) => {
      logger.report(category, msg.title, msg.description);
    }
  );

  return logger;
};
