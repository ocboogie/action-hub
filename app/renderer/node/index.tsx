import * as joi from "joi";
import * as React from "react";

export interface INode {
  name: string;
  component: React.SFC<any> | React.ComponentClass<any>; // Might be the wrong type
  argSchema: joi.ObjectSchema;
}
