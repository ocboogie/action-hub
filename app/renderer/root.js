import React from "react";

export default <h1>We out here</h1>;
