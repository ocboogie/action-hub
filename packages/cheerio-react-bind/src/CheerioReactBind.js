import React from "react";
import PropTypes from "prop-types";

export default class CheerioReactBind extends React.Component {
  constructor(props) {
    super(props);
    this.props.$elem.data("update", this.update.bind(this));
    this.update(true);
  }

  update(first = false) {
    const children = [];
    const $children = this.props.$elem.children();
    if ($children.length > 0) {
      this.props.$elem.children().each((index, elem) => {
        const $child = this.props.$(elem);
        children.push(
          <CheerioReactBind
            key={index}
            $={this.props.$}
            $elem={$child}
            tags={this.props.tags}
          />
        );
      });
    } else {
      children.push(this.props.$elem.text());
    }
    const state = {
      attributes: this.props.$elem.attr(),
      children
    };
    if (first) {
      this.state = state;
    } else {
      this.setState(state);
    }
  }

  render() {
    const { $elem } = this.props;
    const tagName = $elem.prop("tagName").toLowerCase();
    if (!Object.prototype.hasOwnProperty.call(this.props.tags, tagName)) {
      // TODO
      return <div />;
    }
    const Component = this.props.tags[tagName];
    return (
      <Component {...this.state.attributes}>{this.state.children}</Component>
    );
  }
}

/* eslint-disable react/require-default-props, consistent-return */
CheerioReactBind.propTypes = {
  $: (props, propName) => {
    const $ = props[propName];
    if (!$) {
      return new Error('Prop type "$" is required for CheerioReactBind');
    }
    if (!$.root) {
      return new Error(
        'Prop type "$" must be a valid Cheerio object for CheerioReactBind'
      );
    }
  },
  $elem: (props, propName) => {
    const $elem = props[propName];
    if (!$elem) {
      return new Error('Prop type "$elem" is required for CheerioReactBind');
    }
    if (!$elem.data || !$elem.prop) {
      return new Error(
        'Prop type "$" must be a valid Cheerio element for CheerioReactBind'
      );
    }
  },
  // eslint-disable-next-line react/forbid-prop-types
  tags: PropTypes.object.isRequired
};
/* eslint-enable react/require-default-props, consistent-return */
