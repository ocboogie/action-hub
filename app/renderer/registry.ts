import { INode } from "./node";

export interface IRegistry {
  nodes: {
    [nodeName: string]: INode;
  };
}
