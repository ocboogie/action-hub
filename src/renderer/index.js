import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux'
import { Route } from 'react-router'
import { ConnectedRouter, push } from 'react-router-redux';
import { webFrame } from 'electron';

import Root from './components/Root';
import { history, configureStore } from './store/configureStore';
import HomePage from './components/HomePage';
import { initNode } from './actions/node';

const store = configureStore();

// To make sure you can't zoom in
webFrame.setVisualZoomLevelLimits(1, 1);
webFrame.setLayoutZoomLevelLimits(1, 1);

const testNode = {
    "type": "grid",
    "value": [
        {
            "type": "button",
            "value": {
                "type": "node",
                "text": "games",
                "value": {}
            }
        },
        {
            "type": "button",
            "value": {
                "type": "node",
                "text": "apps",
                "value": {}
            }
        }
    ]
}

store.dispatch(initNode(testNode))

render(
    <Root store={store} history={history} />,
    document.getElementById('root')
);

store.dispatch(push("/"));
