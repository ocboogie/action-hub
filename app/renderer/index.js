import ReactDOM from "react-dom";

import root from "./root";

ReactDOM.render(root, document.getElementById("root"));
