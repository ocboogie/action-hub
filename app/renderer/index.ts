// tslint:disable-next-line no-implicit-dependencies
import { ipcRenderer } from "electron";
import * as ReactDom from "react-dom";

import Logger from "../common/Logger";
import createLogger from "./createLogger";
import root from "./root";

const { config, logger: loggerObj } = ipcRenderer.sendSync("get-data");

const logger = createLogger(loggerObj);

declare global {
  // tslint:disable-next-line interface-name
  interface Window {
    logger: Logger;
    config: object;
  }
}

window.logger = logger;
window.config = config;

ReactDom.render(root, document.getElementById("root"));
