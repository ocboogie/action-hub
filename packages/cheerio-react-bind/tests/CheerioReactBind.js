/* eslint react/prop-types: "off" */
import React from "react";
import Cheerio from "cheerio";
import renderer from "react-test-renderer";

import CheerioReactBind from "../src/index";

const tags = {
  div: ({ children }) => <div>{children}</div>,
  h1: ({ children }) => <h1>{children}</h1>,
  p: ({ children }) => <p>{children}</p>
};

const $ = Cheerio.load(
  `
  <div>
    <p>Paragraph</p>
  </div>
`,
  { xmlMode: true }
);

test("modifying the Cheerio dom updates the React dom with custom tags", () => {
  const component = renderer.create(
    <CheerioReactBind tags={tags} $elem={$("div").first()} $={$} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  $("div").append(`<div>test</div>`);
  $("div").data("update")();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  $("p").text("Paragraph test");
  $("p").data("update")();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
