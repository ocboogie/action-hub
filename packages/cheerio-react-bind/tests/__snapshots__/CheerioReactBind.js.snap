// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modifying the Cheerio dom updates the React dom with custom tags 1`] = `
<div>
  <p>
    Paragraph
  </p>
</div>
`;

exports[`modifying the Cheerio dom updates the React dom with custom tags 2`] = `
<div>
  <p>
    Paragraph
  </p>
  <div>
    test
  </div>
</div>
`;

exports[`modifying the Cheerio dom updates the React dom with custom tags 3`] = `
<div>
  <p>
    Paragraph test
  </p>
  <div>
    test
  </div>
</div>
`;
